gwt-ol3-client/src/main/java/ol/MapOptions.java

```diff
@@ -32,49 +35,110 @@
 @JsType(isNative = true, namespace = JsPackage.GLOBAL, name = "Object")
 public class MapOptions implements Options {
 
+    /**
+     * @param controls Controls initially added to the map. If not specified, <code>ol.control.defaults()</code> is used.
+     */
     @JsProperty
-    public native double getPixelRatio();
+    public native void setControls(Collection<Control> controls);
 
     @JsProperty
-    public native String getTarget();
+    public native Collection<Control> getControls();
 
+	/**
+	 * @param pixelRatio The ratio between physical pixels and device-independent pixels (dips) on the device.
+	 * If <code>undefined</code> then it gets set by using <code>window.devicePixelRatio</code>.
+	 */
     @JsProperty
-    public native void setTarget(String target);
+    public native void setPixelRatio(Double pixelRatio);
 
     @JsProperty
-    public native void setLayers(Collection<Base> layers);
+    public native Double getPixelRatio();
+
+	/**
+	 * Interactions that are initially added to the map. If not specified,
+	 * ol.interaction.defaults() is used.
+	 *
+	 * @param interactions
+	 */
+	@JsProperty
+	public native void setInteractions(Collection<Interaction> interactions);
 
     @JsProperty
-    public native View getView();
+    public native void setInteractions(Interaction[] interactions);
```

Comment:
I know that but this answers not my question ;-) Are you using Arrays in your client code if it is not neccessary? From my point of view we don't need this. The user of the API have to make one unneccessary choice more. Besides that there will be probably problems if you use an array and later call the getter which will return a collection. 

This is a point of the ol-api I don't like (from my point of view poor API design). You can use two or more types for one parameter which will not be checked and at every place you want to use the property you have to check the type 1000 times.